src/qibolab/instruments/qblox/controller.py

```diff
@@ -26,33 +27,39 @@ class QbloxController(Controller):
         modules (dict): A dictionay with the qblox modules connected to the experiment.
     """
 
-    def __init__(self, name, cluster, modules):
+    def __init__(self, name, address: str, modules):
         """Initialises the controller."""
-        super().__init__(name=name, address="")
+        super().__init__(name=name, address=address)
         self.is_connected = False
-        self.cluster: Cluster = cluster
+        self.cluster: QbloxCluster = None
         self.modules: dict = modules
+        self._reference_clock = REFERENCE_CLOCK
         signal.signal(signal.SIGTERM, self._termination_handler)
 
     def connect(self):
         """Connects to the modules."""
 
         if self.is_connected:
             return
-        try:
-            self.cluster.connect()
-            for name in self.modules:
-                self.modules[name].connect()
-            self.is_connected = True
-        except Exception as exception:
-            raise_error(
-                RuntimeError,
-                "Cannot establish connection to " f"{self.modules[name]} module. " f"Error captured: '{exception}'",
-            )
-            # TODO: check for exception 'The module qrm_rf0 does not have parameters in0_att' and reboot the cluster
-
-        else:
-            log.info("QbloxController: all modules connected.")
+        for attempt in range(3):
```

Comment:
We were discussing with @stavros11 that attempts might not be required.

Let's be transparent: if the user asks to connect, we will try once. If it fails that's it.
The user can always attempt again on its side (and how many times he wants).

Reply:
I agree, I kept this attempts in order to be more similar to the previous implementation but I would like to delete them. Also while testing I saw that the connection happens always at first time.